src/metabase/driver/clickhouse_qp.clj

```diff
@@ -461,3 +446,13 @@
 (defmethod unprepare/unprepare-value [:clickhouse ZonedDateTime]
   [_ t]
   (format "'%s'" (t/format "yyyy-MM-dd HH:mm:ss.SSSZZZZZ" t)))
+
+;; See https://github.com/ClickHouse/metabase-clickhouse-driver/issues/196
+(def ^:private int-base-types [:type/Integer :type/BigInteger])
+(defmethod sql.params.substitution/align-temporal-unit-with-param-type :clickhouse
+  [_ field param-type]
+  ;; Required for working with integer timestamps
+  ;; See `metabase.query-processor-test.alternative-date-test/substitute-native-parameters-test`
+  (let [base-type (:base-type field)]
```

Comment:
no tests need to be changed for this and IP6?

Reply:
No, `metabase.query-processor-test.alternative-date-test/substitute-native-parameters-test` passes now.